med-inventory/src/components/purchase-medicine/purchase-layout/purchaseLayout.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from "react";
+import Card from "@material-ui/core/Card";
+import axios from "axios";
+import CardContent from "@material-ui/core/CardContent";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import { AgGridReact } from "ag-grid-react";
+import "ag-grid-community/dist/styles/ag-grid.css";
+import "ag-grid-community/dist/styles/ag-theme-balham.css";
+import "./purchaseLayout.css";
+import NumericEditor from "./numericEditor.jsx";
+import PreviewPurchase from "./previewPurchase";
+
+class purchaseLayout extends Component{
+  constructor(){
+    super();
+    this.gridApi={};
+    this.gridOptions = {
+      suppressCellSelection: true,
+      suppressPropertyNamesCheck: true,
+      suppressRowClickSelection: true,
+      accentedSort: true,
+      enableBrowserTooltips: true,
+      suppressPreventDefaultOnMouseWheel: true,
+      suppressDragLeaveHidesColumns: true,
+      headerHeight: 46
+    }
+
+    this.state = {
+      columnDefs: [
+        {
```

Comment:
Since this state has gotten bigger, can we extract the ag-grid config as a function that returns the config or move it to a configuration file?